### PR TITLE
Fix a few multiple enumerations

### PR DESCRIPTION
**Pull request by goremykin**
This PR suppresses one multiple enumeration warning and fixes one real multiple enumeration

**Changes**
See comments below

### CODE REVIEW

File: Emby.Server.Implementations/Library/LibraryManager.cs [hunk 1]
```diff
@@ -358,7 +358,7 @@ public void DeleteItem(BaseItem item, DeleteOptions options, BaseItem parent, bo
 
             var children = item.IsFolder
                 ? ((Folder)item).GetRecursiveChildren(false)
-                : Enumerable.Empty<BaseItem>();
+                : Array.Empty<BaseItem>();
```
Comment (goremykin) [inline Emby.Server.Implementations/Library/LibraryManager.cs:361]: Here I just suppress false multiple enumeration warning. So now `children` is an `IList`

File: MediaBrowser.Providers/Manager/ProviderManager.cs [hunk 2]
```diff
@@ -807,13 +807,11 @@ private bool IsSaverEnabledForItem(IMetadataSaver saver, BaseItem item, LibraryO
             where TLookupType : ItemLookupInfo
         {
             var results = await provider.GetSearchResults(searchInfo, cancellationToken).ConfigureAwait(false);
-
-            foreach (var item in results)
```
Comment (goremykin) [inline MediaBrowser.Providers/Manager/ProviderManager.cs:811]: Here we have a bug. If `provider.GetSearchResults` implementation returns lazy enumeration (like `MusicBrainzArtistProvider` for example), we lose `SearchProviderName`

Comment (cvium) [reply MediaBrowser.Providers/Manager/ProviderManager.cs:811]: Why not remove the method? It's a private method used in only one place.

**Review by cvium — APPROVED** [review]
LGTM

**Review by Bond-009 — APPROVED** [review]
Weird, multiple enumerations should cause errors when building debug mode :thinking:
What warnings did you get CA1851 or something else?

**Comment by goremykin** [comment]
@Bond-009, I received these warnings from the jetbrains rider's analyzer.
There is even one more [here](https://github.com/jellyfin/jellyfin/blob/1f083c0de2a1be669c2af7e77cd4e7313ee8f449/RSSDP/IEnumerableExtensions.cs#L21). But I haven't come up with a good solution yet.

**Review by crobibero — APPROVED** [review]

**Pull request merged**